### DISCUSSION TITLE
Input: How to differentiate ENTER pressed from focusOut

### DISCUSSION THREAD

**Discussion by maitefdupont** [UI5/webcomponents#6920] [Q&A]
According to the [Input Docs](https://sap.github.io/ui5-webcomponents-react/?path=/docs/inputs-input--docs), **onchange** is triggered when ENTER is pressed or on focus out.

In reality these two events are very different. In case of focusout, the user may still be in the middle of typing and will come back. When ENTER is pressed, it should signal that work is done.
My question is, how to differentiate the two?

Many thanks for the support!

**Comment by MarcusNotheis** [comment]
Thanks for reaching out! As the Input component is developed in the UI5 Web Components repository, I'll forward this discussion to their repository.

**Comment by qasic** [comment]
If you want to only capture ENTER events, you can use the keyup event on the element itself.

Here's some sample code:

```
const query = document.getElementById('query');
query?.addEventListener('keyup', (keyEvent) => {
  if (keyEvent.code.includes('Enter')) {
    // ... handle ENTER workflow ...
  }
});
```

Here's the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/Element/keyup_event) for the event.
